engine/src/debug.cpp

```diff
@@ -612,23 +611,25 @@ void MCB_parsewatches(MCExecContext& ctxt, MCStringRef p_input)
 				t_object . object != nil)
 			{
 				Watchvar *t_new_watches;
-				t_new_watches = (Watchvar *)realloc(MCwatchedvars, sizeof(Watchvar) * (MCnwatchedvars + 1));
-				if (t_new_watches != nil)
-				{
-					MCwatchedvars = t_new_watches;
-					if (t_resolved_object)
-						MCwatchedvars[MCnwatchedvars] . object = t_object . object;
-					else
-						MCwatchedvars[MCnwatchedvars] . object = nil;
-					
-					if (MCStringGetLength(*t_hname) != 0)
-						/* UNCHECKED */ MCNameCreate(*t_hname, MCwatchedvars[MCnwatchedvars] . handlername);
-					else
-						MCwatchedvars[MCnwatchedvars] . handlername = nil;
-					/* UNCHECKED */ MCNameCreate(*t_vname, MCwatchedvars[MCnwatchedvars] . varname);
-					MCwatchedvars[MCnwatchedvars] . expression = MCValueRetain(*t_express);
-					MCnwatchedvars++;
-				}
+                if (!MCMemoryReallocate(MCwatchedvars, sizeof(Watchvar) * (MCnwatchedvars + 1), t_new_watches))
+                {
+                    t_success = false;
+                    break;
+                }
+                
+                MCObjectHandle t_objecthandle;
+                MCNewAutoNameRef t_handlername;
+                MCNewAutoNameRef t_varname;
+                
+                if (t_resolved_object)
+                    t_objecthandle = t_object.object;
+                
+                /* UNCHECKED */ MCNameCreate(*t_hname, &t_handlername);
+                /* UNCHECKED */ MCNameCreate(*t_vname, &t_varname);
```

Comment:
Please check these! It should be pretty straightforward because you already have a `t_success` local to hand.